paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java

```diff
@@ -1104,6 +1064,97 @@ CommitResult tryCommitOnce(
                 baseDataFiles);
     }
 
+    public void compactManifest() {
+        Snapshot latestSnapshot = snapshotManager.latestSnapshot();
+
+        if (latestSnapshot == null) {
+            return;
+        }
+
+        List<ManifestFileMeta> mergeBeforeManifests =
+                manifestList.readDataManifests(latestSnapshot);
+
+        List<ManifestFileMeta> mergeAfterManifests =
+                ManifestFileMerger.merge(
+                        mergeBeforeManifests,
+                        manifestFile,
+                        manifestTargetSize.getBytes(),
+                        1,
+                        manifestFullCompactionSize.getBytes(),
```

Comment:
should be 1?

Reply:
As a procedure, if you trigger this, maybe merge all the deleted file is the best choise.